src/csharp/Microsoft.Spark.E2ETest/UdfTests/BroadcastTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Spark.Sql;
+using static Microsoft.Spark.Sql.Functions;
+using Xunit;
+
+namespace Microsoft.Spark.E2ETest.UdfTests
+{
+    [Serializable]
+    public class BroadcastTestClass
+    {
+        public int intValue { get; set; }
+        public string stringValue { get; set; }
+        public double doubleValue { get; set; }
+        public bool boolValue { get; set; }
+
+        public BroadcastTestClass(int _int, string _string, double _double, bool _bool)
+        {
+            intValue = _int;
+            stringValue = _string;
+            doubleValue = _double;
+            boolValue = _bool;
+        }
+    }
+
+    [Collection("Spark E2E Tests")]
+    public class BroadcastTests
+    {
+        private readonly SparkSession _spark;
+        private readonly DataFrame _df;
+
+        public BroadcastTests(SparkFixture fixture)
+        {
+            _spark = fixture.Spark;
+            var data = new List<string>(new string[] { "Alice", "Bob" });
+            _df = _spark.CreateDataFrame(data);
+        }
+
+        /// <summary>
+        /// Function tests Broadcast support by using a single broadcast variable in a UDF.
+        /// </summary>
+        [Fact]
+        public void TestSingleBroadcastWithoutEncryption()
+        {
+            var objectToBroadcast = new BroadcastTestClass(
+                1,
+                "testing first broadcast",
+                1.1,
+                true);
+
+            Broadcast bc = _spark.SparkContext.Broadcast(objectToBroadcast);
+
+            Func<Column, Column> testBroadcast = Udf<string, string>(
+                row => row + " is " + 
+                ((BroadcastTestClass)bc.Value()).stringValue + 
+                ", " + ((BroadcastTestClass)bc.Value()).intValue + 
+                ", " + ((BroadcastTestClass)bc.Value()).doubleValue + 
+                " => " + ((BroadcastTestClass)bc.Value()).boolValue);
+
+            string[] expected = new[] {
+                "Alice is testing first broadcast, 1, 1.1 => True",
+                "Bob is testing first broadcast, 1, 1.1 => True" };
```

Comment:
Should we make the test case more readable?

Reply:
How about something like this:
dataframe -> "row1", "row2" (2 rows)
udf row1 -> "row1: 1, test, 1.0, True" (where 1(int), "test" (string), 1.0 (double), True (bool) are class members, and similar for row 2)
Please feel free to share other suggestions. Thanks!

Reply:
Please see comments below.